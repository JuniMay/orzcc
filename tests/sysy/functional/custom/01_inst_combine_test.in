2934 -2467
