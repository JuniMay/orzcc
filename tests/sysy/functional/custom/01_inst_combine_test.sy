void error() {
  putch(101);
  putch(114);
  putch(114);
  putch(111);
  putch(114);
  putch(10);
}

void ok() {
  putch(111);
  putch(107);
  putch(10);
}

void assert(int cond) {
  if (!cond) {
    error();
  } else {
    ok();
  }
}

int main(){
    int shl=16;
    int negshl = -8;

    int li;     //2934
    int negli;  //-2467

    int cli = 2934;
    int cnegli = -2467;

    li = getint();
    negli = getint();


    assert(li*shl - cli*shl);
    assert(li/shl - cli/shl);
    assert(li%shl - cli%shl);
    putch(10);

    assert(negli*shl - cnegli*shl);
    assert(negli/shl - cnegli/shl);
    assert(negli%shl - cnegli%shl);
    putch(10);

    assert(li*negshl - cli*negshl);
    assert(li/negshl - cli/negshl);
    assert(li%negshl - cli%negshl);
    putch(10);

    assert(negli*negshl - cnegli*negshl);
    assert(negli/negshl - cnegli/negshl);
    assert(negli%negshl - cnegli%negshl);
    putch(10);

    return 0;
}