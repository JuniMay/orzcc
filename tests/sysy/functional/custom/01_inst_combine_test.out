error
error
error

error
error
error

error
error
error

error
error
error

0
